### PR TITLE
Update service provider registration

### PR DESCRIPTION
**Pull request by kshtjvats**

### CODE REVIEW

File: api/src/main/java/com/community/api/endpoint/avisoft/controller/Account/AccountEndPoint.java [hunk 1]
```diff
@@ -8,6 +8,7 @@
 import com.community.api.services.*;
```
Comment (raman-kumar-avisoft) [inline api/src/main/java/com/community/api/endpoint/avisoft/controller/Account/AccountEndPoint.java]: Not import using *
only import the required classes.
image

File: api/src/main/java/com/community/api/endpoint/avisoft/controller/Customer/CustomerEndpoint.java [hunk 2]
```diff
@@ -245,14 +254,22 @@ else if(customerDetails.getFirstName() == null || customerDetails.getLastName()
                 return ResponseService.generateErrorResponse("List of Failed validations : "+errorMessages.toString(),HttpStatus.BAD_REQUEST);
             em.merge(customCustomer);
             return ResponseService.generateSuccessResponse("User details updated successfully : ",sharedUtilityService.breakReferenceForCustomer(customer), HttpStatus.OK);
-
+*/
```
Comment (raman-kumar-avisoft) [inline api/src/main/java/com/community/api/endpoint/avisoft/controller/Customer/CustomerEndpoint.java]: Remove commented code
image

File: api/src/main/java/com/community/api/endpoint/serviceProvider/ServiceProviderEntity.java [hunk 3]
```diff
@@ -2,12 +2,16 @@
...
 import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
 import lombok.AllArgsConstructor;
 import lombok.Getter;
 import lombok.NoArgsConstructor;
 import lombok.Setter;
+import org.hibernate.annotations.Fetch;
+import org.hibernate.annotations.FetchMode;
 
 import javax.persistence.*;
 import javax.validation.constraints.*;
```
Comment (raman-kumar-avisoft) [inline api/src/main/java/com/community/api/endpoint/serviceProvider/ServiceProviderEntity.java:26]: Again avoid importing all the classes just import the respective class you need.
image

File: api/src/main/java/com/community/api/endpoint/serviceProvider/ServiceProviderEntity.java [hunk 4]
```diff
@@ -35,9 +39,10 @@ public class ServiceProviderEntity  {
     private String last_name;
     //@TODO-countryCode to country_code for both customer and service provider
     private String country_code;
+
     private String father_name;
 
-    private Date date_of_birth;
+    private String date_of_birth;
 
     private String aadhaar_number;
```
Comment (raman-kumar-avisoft) [inline api/src/main/java/com/community/api/endpoint/serviceProvider/ServiceProviderEntity.java:56]: Use camelcase in varible naming and use annotation @jsonProperty for snake_casing in postman.
As in java camelcasing is used globally don't mix both casing.
image

File: api/src/main/java/com/community/api/services/DocumentStorageService.java [hunk 5]
```diff
@@ -187,14 +187,16 @@ public void saveAllDocumentTypes() {
...
+                /*new DocumentType(14,"MATRICULATION", "Completed secondary education or equivalent"),
                 new DocumentType( 15,"INTERMEDIATE", "Completed higher secondary education or equivalent"),
                 new DocumentType(16,"BACHELORS", "Completed undergraduate degree program education "),
                 new DocumentType(17,"MASTERS", "Completed postgraduate degree program education"),
                 new DocumentType( 18,"DOCTORATE", "Completed doctoral degree program education"),
                 new DocumentType(19,"DOMICILE", "The permanent home or principal residence of a person."),
-                new DocumentType( 20,"HANDICAPED", "An outdated term for individuals with physical or mental disabilities; \"person with a disability\" is preferred today"),
-       
+                new DocumentType( 20,"HANDICAPED", "An outdated term for individuals with physical or mental disabilities; \"person with a disability\" is preferred today"),*/
+                new DocumentType(21,"C-FORM-PHOTO", "A C Form photo is a standardized ID photo for official documents."),
```
Comment (raman-kumar-avisoft) [inline api/src/main/java/com/community/api/services/DocumentStorageService.java:197]: remove commented code

File: api/src/main/java/com/community/api/services/ServiceProvider/ServiceProviderServiceImpl.java [hunk 6]
```diff
@@ -21,13 +24,13 @@
 import org.springframework.web.bind.annotation.RequestBody;
 import org.springframework.web.client.HttpClientErrorException;
 
-import javax.persistence.EntityManager;
-import javax.persistence.PersistenceContext;
-import javax.persistence.Query;
-import javax.persistence.TypedQuery;
+import javax.persistence.*;
```
Comment (raman-kumar-avisoft) [inline api/src/main/java/com/community/api/services/ServiceProvider/ServiceProviderServiceImpl.java]: use single class import

File: api/src/main/java/com/community/api/endpoint/avisoft/controller/Customer/CustomerEndpoint.java [hunk 7]
```diff
@@ -156,14 +156,23 @@ public ResponseEntity<?> retrieveCustomerById(@RequestParam Long customerId) {
...
             if (customerService == null) {
                 return ResponseService.generateErrorResponse("Customer service is not initialized.", HttpStatus.INTERNAL_SERVER_ERROR);
             }
-
-            CustomCustomer customCustomer = em.find(CustomCustomer.class, customerId);
+            CustomCustomer customerDetails=entityManager.find(CustomCustomer.class,customerId);
+            Set<String> fieldNames = details.keySet();
+            for (String fieldName : fieldNames) {
+                if (isFieldPresent(CustomCustomer.class, fieldName)) {
+                    System.out.println(fieldName + " is a field of CustomCustomer.");
```
Comment (Anilkant8576) [inline api/src/main/java/com/community/api/endpoint/avisoft/controller/Customer/CustomerEndpoint.java]: remove print statement

**Pull request merged**